tidy3d/plugins/mode/mode_solver.py

```diff
@@ -71,6 +82,30 @@ def freqs_not_empty(cls, val):
             raise ValidationError("ModeSolver 'freqs' must be a non-empty tuple.")
         return val
 
+    @pydantic.validator("compute_group_index")
+    def assign_default_on_true(cls, val):
+        """Assing the default fractional frequency step value if not provided."""
+        if val is True:
+            return GROUP_INDEX_STEP
+        return val
+
+    def _post_init_validators(self):
+        if self.compute_group_index >= 1:
```

Comment:
might be better to throw this in the `assign_default_on_true` validator and it could be caught earlier. For example
```py
@pydantic.validator("compute_group_index")
def assign_default_on_true(cls, val):
    """Assing the default fractional frequency step value if not provided."""
    if val is True:
        val = GROUP_INDEX_STEP
    if val >= 1:
        raise ValidationError("Argument 'compute_group_index' must be less than 1.")
    return val
```